server/controllers/stock/reports/stock/value.js

```diff
@@ -30,21 +30,109 @@ async function reporting(_options, session) {
+      gl1.is_exit, gl1.docRef, gl1.valueStock, gl1.quantityIn, (@cumquant := gl1.quantity + @cumquant) AS quantityStock,
+      (@cumsum := gl1.quantityIn + @cumsum) AS quantityCum, (@cumtotal := gl1.valueStock + @cumtotal) AS cumtotal,
+      (@cumOut := gl1.quantityOut + @cumOut) AS cumOut, gl1.created_at
+      FROM (
+        SELECT inv.uuid AS inventory_uuid, inv.text AS inventory_text, sm.date, IF(sm.is_exit, (sm.quantity * (-1)),
+        sm.quantity) AS quantity, sm.unit_cost, sm.is_exit, sm.flux_id, map.text AS docRef, sm.created_at,
+        IF (sm.is_exit, 0, (sm.quantity * sm.unit_cost)) AS valueStock, IF (sm.is_exit, 0, sm.quantity) AS quantityIn,
+        IF (sm.is_exit, sm.quantity, 0) AS quantityOut
+        FROM stock_movement AS sm
+        JOIN lot AS l ON l.uuid = sm.lot_uuid
+        JOIN inventory AS inv ON inv.uuid = l.inventory_uuid
+        JOIN document_map AS map ON map.uuid = sm.document_uuid
+        WHERE inv.uuid = ? AND sm.depot_uuid = ? AND DATE(sm.date) <= DATE(?)
+        ORDER BY DATE(sm.date), sm.created_at  ASC
+      ) AS gl1, (SELECT @cumquant := 0)a, (SELECT @cumsum := 0) AS b, (SELECT @cumtotal := 0)c, (SELECT @cumOut := 0)d
+    `;
+
+    transaction
+      .addQuery(sqlGetValueByMovement, [db.bid(inventory.inventory_uuid), db.bid(options.depot_uuid), options.dateTo]);
+  });
+
+  const inventoriesValues = await transaction.execute();
+  inventoriesValues.forEach(inventory => {
+    let unitCost = 0;
+    let lastCumValue = 0;
+
+    inventory.forEach(item => {
+      item.oldUNITCOST = unitCost;
+      if (item.is_exit) {
+        item.newQuantityCum = item.quantityCum - item.cumOut;
+        item.newCumValue = item.quantityStock * unitCost;
+        item.stockUnitCost = item.newQuantityCum ? item.newCumValue / item.newQuantityCum : unitCost;
+
+        unitCost = item.stockUnitCost;
+        lastCumValue = item.newCumValue;
+      } else if (!item.is_exit) {
+        item.newQuantityCum = item.quantityCum - item.cumOut;
+        item.newCumValue = item.valueStock + lastCumValue;
+        item.stockUnitCost = item.newQuantityCum ? item.newCumValue / item.newQuantityCum : unitCost;
+
+        unitCost = item.stockUnitCost;
+        lastCumValue = item.newCumValue;
+      }
+    });
+  });
+
+  let stockTolal = 0;
+
+  stockValues.forEach(inventory => {
+    inventory.movements = [];
+    inventoriesValues.forEach(item => {
+      const lastMovement = item[item.length - 1];
+
+      if (inventory.inventory_uuid === lastMovement.inventory_uuid) {
+        inventory.stockQtt = lastMovement.quantityStock;
+        inventory.stockUnitCost = lastMovement.stockUnitCost;
+        inventory.stockValue = lastMovement.newCumValue;
+
+        stockTolal += inventory.stockValue;
+      }
+    });
+  });
 
   const stockValueElements = options.exclude_zero_value
-    ? stockValues[0].filter(item => item.stockValue > 0) : stockValues[0];
+    ? stockValues.filter(item => item.stockValue > 0) : stockValues;
```

Comment:
Is there anyway we can apply this filter higher in logic?  Preferably in the SQL.  It would probably improve performance.

Reply:
it will be difficult seeing that the products we see the stock shortage of a product after on the last line, after having noted all the entries and exits from Stock